src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -431,27 +431,6 @@ void Terminal::_WriteBuffer(const std::wstring_view& stringView)
         {
             proposedCursorPosition.Y++;
         }

```

Comment:
Yep. I believe the only reason it's still needed is for the `TerminalDispatch::LineFeed` implementation here:

https://github.com/microsoft/terminal/blob/94f4db32cdbc41fc9f3234f5c35554f230aab7fc/src/cascadia/TerminalCore/TerminalDispatch.cpp#L85-L91

Ideally I think it would be replaced with a call to something in the `ITerminalApi` interface, similar to the way the `AdaptDispatch` implementation calls `ConGetSet::PrivateLineFeed`. But I was hoping that would eventually work itself out as part of the `TerminalDispatch`/`AdaptDispatch` unification in #3849.

If it's holding up your #780 stream writer, though, it may be worth doing something about this sooner rather than later, even if it is a temporary hack.

Reply:
I keep changing my mind on what I want the stream writer to be because `TextBuffer::Write` doesn't seem far off from it right now, especially with all the C0 stuff coming out.

I think for consistency sake, we should probably take \n out too into a temporary thing. Then all this method will have left is the cursor handling and viewport scrolling.

Reply:
I'm not sure this is what you had in mind, but I had to split the cursor movement code out of `Terminal::_WriteBuffer` into a separate method so it could be reused in the linefeed implementation (see [`7f0bcad`](https://github.com/microsoft/terminal/commit/7f0bcad72bba5e131ba94aab84efdc6a911247e5?w=1)  without whitespace diffs), otherwise there'd be a lot of code duplication.

I then added a `CursorLineFeed` method in `TerminalApi` which calls the above method to handle the actual cursor movement, and got rid of the linefeed handling in `_WriteBuffer`  (see [`da0924b`](https://github.com/microsoft/terminal/commit/da0924b43df07f2d5831f38090f83949575659fb?w=1), again easier to follow without whitespace diffs).

I'm not positive this is the best long-term interface, but for now at least it's more in line with the `AdaptDispatch` implementation.

Reply:
OK, I'll look. Thank you for bearing with me.

Reply:
Nah, that's good enough for now. Thanks. Let's go with what you said further below and just keep progressively mushing on it until we deem it minimally offensive.